analyzers/tests/SonarAnalyzer.UnitTest/Rules/ParameterValidationInYieldShouldBeWrappedTest.cs

```diff
@@ -44,6 +44,15 @@ public class ParameterValidationInYieldShouldBeWrappedTest
             builder.AddPaths("ParameterValidationInYieldShouldBeWrapped.CSharp11.cs")
                 .WithOptions(ParseOptionsHelper.FromCSharp11)
                 .Verify();
+#endif
+
+#if NET8_0_OR_GREATER
```

Comment:
This test will never be executed, since we build the UTs against .NET7. you should put it together with the 10 and 11 tests inside #if NET.

Reply:
The test requires `System.Runtime.CompilerServices.InlineArrayAttribute`, that only exists in .NET 8.
As far as I know, the only way to run UTs against .NET 8, is to change `TargetFrameworks` to `net48;net8` from `net48;net7.0`, which is what I have done locally, to run this UT.

I haven't included such a change in this PR because .NET 8 hasn't been released yet: my plan was to keep this test disabled, until .NET 8 is officially released, and then switch `SonarAnalyzer.UnitTest.csproj` to .NET 8. At that point the test should be movable into the `#if NET` region.

What do you think?

Reply:
As discussed offline we could try to change the MetadataReferenceFactory and use the .NET 8 system assemblies for these tests.

Reply:
As discussed:
- I have removed the UT from this PR, to be able to deliver it
- I have extracted the issue into a dedicated card: https://trello.com/c/6K0EezU7/21-support-net8-uts-for-s2368-fn-with-inlinearrayattribute
- I am going to try applying the fix identified by @csaba-sagi-sonarsource in that card

Reply:
> I have removed the UT from this PR,

It looks to me that the test is still in this PR. Do I miss something?

Reply:
There actually were two of them. Fixed.